app/user_views.py

```diff
@@ -1,42 +1,75 @@
+        if requested_rehearsal.user_id != current_user.id:
+            abort(403)
+        return func(*args, **kwargs)
+    return wrapper
 
 
-@app.route("/all-rehearsals")
+@app.route("/all-rehearsals", methods=["GET", "POST"])
 @login_required
 def get_all_rehearsals():
-    rehearsals = Rehearsal.query.all()
+    # Check date
+    date_check = dt.datetime.today().date()
+    today = date_check.strftime("%a %d %b, %Y")
+
+    form = OrderForm()
+    order_by = request.args.get("order_by")
+    if not order_by:
+        # Handle the case when the user does not select any option
+        # redirect the user to a default ordering
+        return redirect(url_for("get_all_rehearsals", order_by="desc"))
+    elif order_by not in ["asc", "desc"]:
+        # Handle the case when the user is trying to manipulate the order_by parameter
+        # return an error message
+        flash("Invalid value for the order_by parameter")
+        return redirect(url_for("get_all_rehearsals", order_by="desc"))
+    else:
+
+        if order_by == "desc":
+            order_by_clause = Rehearsal.date.desc()
+        else:
+            order_by_clause = Rehearsal.date.asc()
+        rehearsals = Rehearsal.query.filter_by(user_id=current_user.id)
+        if order_by == "desc":
+            rehearsals = rehearsals.order_by(Rehearsal.date.desc())
+        elif order_by == "created":
+            rehearsals = rehearsals.order_by(Rehearsal.user_id.asc())
+        else:
+            rehearsals = rehearsals.order_by(Rehearsal.date.asc())
+
+    # filter by distinct
+    distinct_groups = db.session.query(Rehearsal.group).filter_by(user_id=current_user.id).distinct().all()
+
     return render_template('user/all_rehearsals.html', all_rehearsals=rehearsals, current_user=current_user,
-                           logged_in=current_user.is_authenticated)
+                           logged_in=current_user.is_authenticated, distinct_groups=distinct_groups, form=form,
+                           order_by=order_by, today=today, date_check=date_check)
 
 
 @app.route("/rehearsal/create", methods=["GET", "POST"])
 @login_required
 def create():
     form = RehearsalForm()
-    if request.method == "POST":
-        req = request.form
-        r_date = req.get("date")
-        group = req.get("group")
-        # Use the datetime module to parse the date string
-        date_time = dt.datetime.strptime(r_date, '%Y-%m-%d')
+    if form.validate_on_submit():
+        user = User.query.filter_by(id=current_user.id).first()
+        r_date = form.date.data
 
-        # Convert the datetime object to a date object
-        date = date_time.date()
+        group = form.group.data
 
         # Insert the row and get the cursor object
-        result = db.session.execute(insert(Rehearsal).values(date=date, group=group))
+        result = db.session.execute(insert(Rehearsal).values(user_id=user.id, date=r_date, group=group))
```

Comment:
Nit: This is a pretty long line. Maybe look at using [Black](https://black.readthedocs.io/en/stable/) soon? Also (as a bonus) pre-commit, so it runs Black automatically.